### ISSUE TITLE
Streamed writing of response

### ISSUE THREAD

**Issue by ypopovych** [ellzey/libevhtp#203]
Hi, ellzey. 
Thanks for cool library!

I have one question and can't find any examples. I wan't to write some data as they produced by my app. 
At this moment I have two threads in app. One for networking and one loop for tasks. On request I will add a task with parameters and need to stream data back. 
I tried to call `evhtp_request_pause` in request cb and then add data using `evhtp_send_reply_chunk*`methods. But this data only buffering, before `evhtp_request_resume` called on request. But if `evhtp_request_pause` not called, request will be deleted before all data sent.

How I can implement streaming?

Thanks, Yehor.

**Comment by ellzey** [comment]
Yes, this is very bad, and I should feel bad about myself. In short this is due to a patch I wrote to fix some http-pipelining issues where you could leak data from the previous request. For a full description of the problem, see my replies here: https://github.com/haiwen/seafile/issues/1119

This is a very serous issue, one that I can't figure out how to work around using Libevent 2.0. But I'm trying (just very busy at the moment, constant interviews, no job, etc)
